packages/roosterjs-editor-dom/lib/selection/Position.ts

```diff
@@ -131,46 +131,51 @@ export default class Position {
      * Get bounding rect of this position
      */
     getRect(): Rect {
-        let document = this.node && this.node.ownerDocument;
+        const document = this.node && this.node.ownerDocument;
 
         if (!document) {
             return null;
         }
 
-        let range = document.createRange();
+        const range = document.createRange();
         range.setStart(this.node, this.offset);
 
         // 1) try to get rect using range.getBoundingClientRect()
-        let rect = normalizeRect(range.getBoundingClientRect());
-
-        if (!rect) {
-            let { node, element, offset } = this.normalize();
-
-            // 2) if current cursor is inside text node, use range.getClientRects() for safari
-            // or insert a SPAN and get the rect of SPAN for others
-            if (Browser.isSafari) {
-                let rects = range.getClientRects();
-                if (rects && rects.length == 1) {
-                    rect = normalizeRect(rects[0]);
-                }
-            } else {
-                if (node.nodeType == NodeType.Text) {
-                    let span = document.createElement('SPAN');
-                    range.setStart(node, offset);
-                    range.collapse(true /*toStart*/);
-                    range.insertNode(span);
-                    rect = normalizeRect(span.getBoundingClientRect());
-                    span.parentNode.removeChild(span);
-                }
+        let rect = range.getBoundingClientRect ? normalizeRect(range.getBoundingClientRect()) : null;
```

Comment:
The function getBoundingClientRect() may not always work, especially when the position is in middle of a text node. So the original code is still required. The only change we need here is to remove the check of isSafari and always try getClientRects(), and then fallback to the code to insert a temp SPAN if it fails.